### ISSUE TITLE
FieldState constructor allow only value

### ISSUE THREAD

**Issue by janpieterz** [formstate/formstate#16]
Given that the FieldState constructor requires a config object that only has one obligatory value (the `TValue`) it would be nice sugar if something like the following would be the constructor syntax'es. This example uses a FieldState<boolean> for simplicity:

(did not test below code so not 100% sure if that would be the best way, possibly `FieldStateConfig` should just be an interface?)
```typescript
class FieldStateConfig {
        value: TValue;
        onUpdate?: (state: FieldState<TValue>) => any;
        autoValidationEnabled?: boolean;
        autoValidationDebounceMs?: number;
}
constructor(config: FieldStateConfig);
constructor(value?: any) {
    if(value ! instanceOf FieldStateConfig) {
        value = new FieldStateConfig(value);
    } 
    //execute default logic    
}
```

If I'm correct, this should be callable by:

```typescript
new FieldState<boolean>(true);
```
and
```typescript
new FieldState<boolean>({
    value: true,
    autoValidationDebounceMs: 250,
    //etc
});
```

**Comment by basarat** [comment]
The api used to be `FieldState({value,onUpdate}); // both compulsory` when the project originated about 2 years ago at my previous gigs and still used `setState` in `onUpdate`. 

It needs to just take `value` with chain functions for `onUpdate` / `autoValidationEnabled` / `autoValidationDebouncedMs`. PR work approved along these lines :rose:

**Comment by basarat** [comment]
Released with `0.18.0` :rose:
